### ISSUE TITLE
Result is not the same as the paper. Is this still under construction?

### ISSUE THREAD

**Issue by ZhangYushan3** [Ferenas/CPN#4]
Hi, I've tested the val set by your pretrained model. But I think the threshold is not set? And the result is about 55 mIoU. So is this repo still under construction?  Which threshold should I set?
Thanks very much!

**Comment by ZhangYushan3** [comment]
Also I did not find the CRF in this repo. Will it be added?

**Comment by SII-Ferenas** [comment]
Hi, the background threshold shall be ground-searched in the ***evaluation_cam.py*** from range 0.2 to 0.6. The ***infer_cam.py*** simply calculates the foreground. According to my experiments, the val set could reach 57.59. Note that the paper used the ***train set***. Certainly, you can add CRF as the post-processing, which may reach about 59 mIoU after fine-tuning the hyper-parameters in CRF.

![eval](https://user-images.githubusercontent.com/55052174/142762548-bc64d73c-7cf5-42b3-8e1d-780df0228cc9.jpg)

**Comment by ZhangYushan3** [comment]
Thanks for your reply. Then I think I get the same result. However in the paper you said that you get a mIoU of 67.8 on validation set of VOC 2012. Could you please tell me how can I get the same result as that one?

Best
Yushan

**Comment by SII-Ferenas** [comment]
I have written the relevant process in README.md, actually, it is a common process in WSSS. You can also refer to IRN and SEAM for more details.

**Comment by ZhangYushan3** [comment]
Hi, is it possible to share the cam result of 67.8 mIoU on validation set? I mean the .npy or .png result. It would be very helpful to me if possible :p

**Comment by ZhangYushan3** [comment]
Or if it's possible to share the trained weights after the common process in WSSS?

**Comment by SII-Ferenas** [comment]
Do you mean weights for DeepLabv1 for the final segmentation?
